File: frontend/src/types/bateria.ts
/** TypeScript interfaces for Bateria entity matching Pydantic schemas. */

import type { BateriaStatus } from './enums';

export interface BateriaResponse {
  id: number;
  identificacao: string;
  drone_id: number | null;
  status: BateriaStatus;
  observacoes: string | null;
  ciclos: number;
  ativo: boolean;
  created_at: string;
  updated_at: string;
}

export interface BateriaCreate {
  identificacao: string;
  drone_id?: number | null;
  status?: BateriaStatus;
  observacoes?: string | null;
}

export interface BateriaUpdate {
  identificacao?: string | null;
  drone_id?: number | null;
  status?: BateriaStatus | null;
  observacoes?: string | null;
}

File: app/schemas/bateria.py
"""Pydantic schemas for Bateria entity."""

from datetime import datetime

from pydantic import BaseModel, Field

from app.models.enums import BateriaStatus


class BateriaBase(BaseModel):
    identificacao: str = Field(..., max_length=120)
    drone_id: int | None = None
    status: BateriaStatus = BateriaStatus.DISPONIVEL
    observacoes: str | None = None


class BateriaCreate(BateriaBase):
    pass


class BateriaUpdate(BaseModel):
    identificacao: str | None = Field(None, max_length=120)
    drone_id: int | None = None
    status: BateriaStatus | None = None
    observacoes: str | None = None


class BateriaResponse(BateriaBase):
    id: int
    ciclos: int
    ativo: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
